src/java/org/apache/cassandra/index/sai/disk/v1/segment/IndexSegmentSearcher.java

```diff
@@ -84,6 +88,20 @@ else if (index.termType().isLiteral())
      */
     public abstract KeyRangeIterator search(Expression expression, AbstractBounds<PartitionPosition> keyRange, QueryContext queryContext) throws IOException;
 
+    /**
+     * Order the rows by the given expression.
+     *
+     * @param exp      the object containing the ordering logic
+     * @param keyRange key range specific in read command, used by ANN index
+     * @param context  to track per sstable cache and per query metrics
+     * @return an iterator of {@link PrimaryKeyWithScore} in descending score order
```

Comment:
nit: I'd throw this on a new line. Just helps separate it visually from the params.